src/math/auditory/auditory.c

```diff
@@ -0,0 +1,214 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Intel Corporation. All rights reserved.
+//
+// Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+
+#include <sof/audio/format.h>
+#include <sof/lib/alloc.h>
+#include <sof/math/auditory.h>
+#include <sof/math/decibels.h>
+#include <sof/math/fft.h>
+#include <sof/math/log.h>
+#include <sof/math/numbers.h>
+#include <ipc/topology.h>
+#include <errno.h>
+#include <stdint.h>
+
+#define ONE_Q16 Q_CONVERT_FLOAT(1, 16)
+#define ONE_Q20 Q_CONVERT_FLOAT(1, 20)
+#define ONE_Q30 Q_CONVERT_FLOAT(1, 30)
+#define TWO_Q29 Q_CONVERT_FLOAT(2, 29)
+
+#define LOG_SCALE_HZMEL_Q26 Q_CONVERT_FLOAT(18.02182669, 26)
+#define LOG_MULT_HZMEL_Q20 Q_CONVERT_FLOAT(1126.9941805389, 20)
+#define ONE_OVER_HZDIV_Q31 Q_CONVERT_FLOAT(1.0 / 700, 31)
+#define ONE_OVER_MELDIV_Q31 Q_CONVERT_FLOAT(1.0 / 1126.9941805389, 31)
+
+#define MEL_MAX_Q2 Q_CONVERT_FLOAT(4358.351140, 2) /* 32767 Hz */
+
+#define LOG2_2P16 Q_CONVERT_FLOAT(16.0, 16) /* log2(2^16) as Q16.16 */
+
+/* For scaling log2 power to log, log10, or dB*/
+#define ONE_OVER_LOG2E_Q29	Q_CONVERT_FLOAT(0.6931471806, 29) /* 1/log2(exp(1))) */
+#define ONE_OVER_LOG2TEN_Q29	Q_CONVERT_FLOAT(0.3010299957, 29) /* 1/log2(10) */
+#define TEN_OVER_LOG2TEN_Q29	Q_CONVERT_FLOAT(3.0102999566, 29) /* 10/log2(10) */
+
+/**
+ * Convert Hz to Mel
+ * Wikipedia https://en.wikipedia.org/wiki/Mel_scale
+ * mel = 1126.9941805389 * log(1 + hz / 700)
+ *
+ * @param hz Frequency as Q16.0 Hz, max 32767 Hz
+ * @return Value as Q14.2 Mel, max 4358.4 Mel
+ */
+int16_t psy_hz_to_mel(int16_t hz)
```

Comment:
qualify for static inline i.e. will it get called a lot in hot code ?

Reply:
No, it's used only initialization. The Mel filterbank apply is the hot part. Did you find other inline candidates?

Reply:
no, but these only came to attention as they were small and simple.